lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -1393,6 +1414,9 @@ module CredentialsInfo
       # Returns the project ID (as a string) on success, or nil on failure.
       def self.project_id
         creds = Google::Auth.get_application_default(LOGGING_SCOPE)
+        if creds.respond_to?(:project_id)
+          return creds.project_id if creds.project_id
+        end
```

Comment:
Should we add a unit test to test the fallback when `project_id` is not present?

Reply:
Our existing tests do exactly that, don't they?

Reply:
Our existing test `test_project_id_from_credentials` test three types of credentials independently. Here we seem to have some order/precedence as well: 1. `creds.project_id`, 2. `creds.issuer`, 3. `creds.client_id`. Not directly related to this PR though. We can defer it till later.

Reply:
Yes, the test data for the newly added test (`new-style-credentials.json`) has a different project encoded in `client_email`, so we *are* testing that order of precedence. As for `issuer` vs `client_id`, I believe the two refer to the same field (but `issuer` is the newer name for it).